go/vt/vttablet/tabletmanager/rpc_backup.go

```diff
@@ -89,9 +92,19 @@ func (tm *TabletManager) Backup(ctx context.Context, logger logutil.Logger, req
 		if err := tm.changeTypeLocked(ctx, topodatapb.TabletType_BACKUP, DBActionNone, SemiSyncActionUnset); err != nil {
 			return err
 		}
+
+		// adding defer to original value in case of any failures
+		defer func() {
+			bgCtx := context.Background()
+			// Change our type back to the original value.
+			// Original type could be primary so pass in a real value for PrimaryTermStartTime
```

Comment:
I kept the original code for reverting to original type down at line# 127. I wanted to remove that block entirely but at the same time I want to make revert part of regular flow as well so we can report error in case of any failure during revert. With defer we won't be able to pop up error to the client. 
Do you think we should remove that revert code from line#127

Reply:
I am in two minds. On one hand, I am not sure if the error message in revert is something that the user cares about at all. Because we retry state change in vttablets, the error could very well be a transient one. On the other hand, ignoring errors is generally a recipe for more problems down the line. Lets see what @deepthi thinks.

Reply:
It doesn't make sense to fail the whole backup if the tablet type revert fails. Let's remove the code lines 127-143.